Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -66,60 +76,68 @@ public final class ViewAnnotationManager {
         if creationOptions.height == nil {
             creationOptions.height = view.bounds.size.height
         }
-        let annotatonView = AnnotationView(view: view, annotationManager: self)
-        try mapboxMap.addViewAnnotation(withId: annotatonView.id, options: options)
-        annotationViewsById[annotatonView.id] = annotatonView
-        annotationIdByView[annotatonView] = annotatonView.id
-        containerView.addSubview(annotatonView)
-        return annotatonView
+
+        let id = String(currentViewId)
+        currentViewId += 1
+
+        view.translatesAutoresizingMaskIntoConstraints = false
+
+        try mapboxMap.addViewAnnotation(withId: id, options: creationOptions)
+        viewsById[id] = view
+        idsByView[view] = id
+        expectedHiddenForId[id] = !(creationOptions.visible ?? true)
+        containerView.addSubview(view)
     }
 
-    /// Remove given `AnnotationView` from the map if it was present.
+    /// Remove given `UIView` from the map if it was present.
     ///
     /// - Parameters:
-    ///   - annotatonView: `AnnotationView` to be removed
+    ///   - view: `UIView` to be removed
     ///
     /// - Throws:
     ///   - `MapError`: errors during the removal of the view
-    public func remove(_ annotatonView: AnnotationView) throws {
-        guard let id = annotationIdByView[annotatonView], let annotatonView = annotationViewsById[id] else {
+    public func remove(_ view: UIView) throws {
+        guard let id = idsByView[view], let annotatonView = viewsById[id] else {
             return
         }
         try mapboxMap.removeViewAnnotation(withId: id)
         annotatonView.removeFromSuperview()
-        annotationViewsById.removeValue(forKey: id)
-        annotationIdByView.removeValue(forKey: annotatonView)
+        viewsById.removeValue(forKey: id)
+        idsByView.removeValue(forKey: annotatonView)
```

Comment:
Under what circumstances would `try mapboxMap.removeViewAnnotation(withId: id)` fail? Would we still want to perform the rest of the method in those situations?

Reply:
Currently it would only fail if it was not found: https://github.com/mapbox/mapbox-gl-native-internal/blob/e0253c010a42b7e57dc17331cb47234e41a428aa/internal/src/mapbox/maps/view_annotation_manager.cpp#L299

Currently it will only happen if the user already removed it. I think I could make the rest of the calls deferred, since we would like to remove it anyway. Unless the implementation changes and the removal could fail for another reason.

Reply:
What about making the method fail silently? Remove `throws` and use `try?`.

Reply:
That's fine, I think it's better if we throw less. 👌